internal/buffer/unbounded_test.go

```diff
@@ -119,3 +119,14 @@ func (s) TestMultipleWriters(t *testing.T) {
 		t.Errorf("reads: %#v, wantReads: %#v", reads, wantReads)
 	}
 }
+
+func (s) TestClose(t *testing.T) {
```

Comment:
Please add top level docstring explaining what this function is testing.

Reply:
Please also mention nothing should be sent on the channel after Close() (i.e. from that put() call)